packages/wonder-blocks-toolbar/components/toolbar.js

```diff
@@ -0,0 +1,206 @@
+
+        // TODO(scottgrant): Replace the SVG with an Icon component once
+        // that's done
+        const fillColor = (style && style.color) || "rgba(33, 36, 44, 0.5)";
+        const closeButton = showClose && (
+            <View onClick={onClose}>
+                <svg role="img" width="24px" height="24px">
+                    <path fill={fillColor} d={dismissIcon} />
+                </svg>
+            </View>
+        );
+
+        // TODO(scottgrant): Handle aria-labelledby appropriately.
+        return (
+            <View
+                style={[
+                    sharedStyles.container,
+                    style,
+                    size === "small" && sharedStyles.small,
+                ]}
+            >
+                <View style={sharedStyles.verticalAlign}>
+                    {closeButton}
+                    {leftContent &&
+                        leftContent.map((content, i) =>
+                            this.renderContent(content, i.toString()),
+                        )}
+                </View>
+                <View style={[sharedStyles.column]}>
+                    <View
+                        style={[
+                            sharedStyles.middleColumn,
+                            sharedStyles.verticalAlign,
+                        ]}
+                    >
+                        <span
+                            id="wb-toolbar-title"
+                            className={css(
+                                sharedStyles.title,
+                                !subtitle && sharedStyles.titleNoSubtitle,
+                            )}
+                        >
+                            {title}
+                        </span>
+                        {subtitle && (
+                            <span className={css(sharedStyles.subtitle)}>
+                                {subtitle}
+                            </span>
+                        )}
+                    </View>
+                </View>
+                <View style={sharedStyles.rightColumn}>
+                    {rightContent &&
+                        rightContent.map((content, i) =>
+                            this.renderContent(content, i.toString()),
+                        )}
+                </View>
+            </View>
+        );
+    }
+}
+
+const sharedStyles = StyleSheet.create({
+    container: {
+        border: "1px solid rgba(33, 36, 44, 0.16)",
+        display: "flex",
+        flexDirection: "row",
+        justifyContent: "space-between",
+        minHeight: 64,
+        padding: 8,
+        position: "relative",
+        width: "100%",
+    },
+    small: {
+        minHeight: 48,
+    },
+    verticalAlign: {
+        display: "flex",
+        flexDirection: "column",
+        justifyContent: "center",
+    },
+    column: {
+        display: "block",
+        position: "absolute",
+        left: "50%",
+        top: "50%",
+        transform: "translate(-50%, -50%);",
+        width: "80%",
```

Comment:
How do we handle collisions between columns?  We might need different styles for the columns depending on what props have been passed in, e.g. if there's a title then the left/right columns are constrained to be a certain width.

Reply:
I'm going to take another look at this. The variable width columns and requirement that the title is right in the middle was a bit of a snag. There are some overlap issues here with long titles, so I'll take a closer look now.